crates/transaction-pool/src/pool/parked.rs

```diff
@@ -295,18 +295,49 @@ impl<T: PoolTransaction> ParkedPool<BasefeeOrd<T>> {
         transactions
     }
 
-    /// Removes all transactions and their dependent transaction from the subpool that no longer
-    /// satisfy the given basefee.
+    /// Removes all transactions from this subpool that can afford the given basefee,
+    /// invoking the provided handler for each transaction as it is removed.
+    ///
+    /// This method enforces the basefee constraint by identifying transactions that now
+    /// satisfy the basefee requirement (typically after a basefee decrease) and processing
+    /// them via the provided transaction handler closure.
+    ///
+    /// Respects per-sender nonce ordering: if the lowest-nonce transaction for a sender
+    /// still cannot afford the basefee, higher-nonce transactions from that sender are skipped.
     ///
     /// Note: the transactions are not returned in a particular order.
-    pub(crate) fn enforce_basefee(&mut self, basefee: u64) -> Vec<Arc<ValidPoolTransaction<T>>> {
+    pub(crate) fn enforce_basefee_with<F>(&mut self, basefee: u64, mut tx_handler: F)
+    where
+        F: FnMut(Arc<ValidPoolTransaction<T>>),
+    {
         let to_remove = self.satisfy_base_fee_ids(basefee as u128);
 
-        let mut removed = Vec::with_capacity(to_remove.len());
         for id in to_remove {
-            removed.push(self.remove_transaction(&id).expect("transaction exists"));
+            if let Some(tx) = self.remove_transaction(&id) {
+                tx_handler(tx);
+            } else {
+                tracing::warn!(
+                    target: "reth::txpool::parked",
+                    ?id,
+                    "Failed to remove transaction during basefee enforcement - possible data structure inconsistency"
+                );
```

Comment:
we can ignore that case here because that invariant is enforced, this was even an expect earlier

Reply:
ahh okay, was afraid that the expect might have cause some issues / wanted to log - but makes sense!

Reply:
done! @mattsse